File: phonebook_copy2/phonebook/phonebook.go

package phonebook

import (
	"phonebook_copy2/binaryTree"
	"strconv"
)

type Phonebook struct
{
	nameTree binaryTree.Bst
	numberTree binaryTree.Bst
}

func (p *Phonebook) Add(number string, name string,address string) {
	p.nameTree.InsertName(name,number,address)
	p.numberTree.InsertNumber(name,number,address)
}

//true used to decide when to print values or not
func (p *Phonebook) Find(value string) {
	if _, err := strconv.Atoi(value); err == nil {
		p.numberTree.FindNumber(value,true)
	} else {
		p.nameTree.FindName(value,true)
	}
}

func (p *Phonebook) Remove(value string) {
	if _, err := strconv.Atoi(value); err == nil {
		name := p.numberTree.FindNumber(value,false)
		p.numberTree.RemoveNumber(value)
		p.nameTree.RemoveName(name)
	} else {
		number := p.numberTree.FindName(value,false)
		p.nameTree.RemoveName(value)
		p.nameTree.RemoveNumber(number)
	}
}

func (p *Phonebook) ShowAll() {
	p.nameTree.ShowBook()
}

File: phonebook_copy1/phonebook.py

from binaryTree import *

class Phonebook():

    def __init__(self):
        self.nameTree = Tree()
        self.numberTree = Tree()

    def add(self,name,number,address):
        self.nameTree.add(name,number,address,"name")
        self.numberTree.add(name,number,address,"number")
    
    def find(self,value):
        if value.isdecimal():
            self.numberTree.find(self.numberTree.root,value,True)        #will return name
        else:
            self.nameTree.find(self.nameTree.root,value,True)       #will return number
    
    def remove(self,value):       
        if value.isdecimal():       #if value is a number 
            node = self.numberTree.find(self.numberTree.root,value,False)   #will be node with all info if not None
            if node is not None:
                name = node.name 
                self.numberTree.remove(self.numberTree.root,value)   #return name
                self.nameTree.remove(self.nameTree.root,name)
            else:
                print("{} not found, cannot be removed from contacts\n".format(value))
        else:
            node = self.nameTree.find(self.nameTree.root,value,False)  #will be node with all info if not None
            if node is not None:
                number = node.number
                self.nameTree.remove(self.numberTree.root,value)   #return name
                self.numberTree.remove(self.nameTree.root,number)
            else:
                print("{} not found, cannot be removed from contacts\n".format(value))
    
    def showAll(self):
        self.nameTree.printTree()        #only one needs to be printed since they are the same

